### PR TITLE
Create acount

### PR DESCRIPTION
**Pull request by EricLarson2020**
Created user model and create user path with corresponding tests

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class UsersController < ApplicationController
+
+  def new
+  end
+
+  def create
+    new_user = User.create(user_params)
+    flash[:success] = "Hello #{new_user.first_name} #{new_user.last_name}, thank you for creating an account!"
```
Comment (EricLarson2020) [inline app/controllers/users_controller.rb:8]: Need to move this to another line, so it does not go over character limit

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  validates :email, uniqueness:true, presence:true
+  validates_presence_of :first_name, :last_name, :password
```
Comment (EricLarson2020) [inline app/models/user.rb:3]: might run into trouble with password here, may need to add param on create

File: config/routes.rb [hunk 3]
```diff
@@ -3,4 +3,7 @@
 
    get "/", to: "welcome#index"
 
+
+   get '/profile', to: "users#show"
```
Comment (EricLarson2020) [inline config/routes.rb:7]: Is this route ok, or did u want a separate profile/dashboard controller instead

File: spec/features/users/new_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    fill_in :last_name, with: last_name
+
+    fill_in :email, with: email
+    fill_in :password, with: password
+    fill_in :password_verification, with: password
+
+    click_on "Create User"
+
+    expect(current_path).to eq("/profile")
+    message = "Hello #{first_name} #{last_name}, thank you for creating an account"
```
Comment (EricLarson2020) [inline spec/features/users/new_spec.rb:23]: This line is to long, need to move it to two lines

File: config/routes.rb (hunk 3 above)
Comment (jpc20) [reply config/routes.rb:7]: Looks good to me.

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+<%= label_tag :first_name %>
+<%= text_field_tag :first_name %>
+
+<%= label_tag :last_name %>
+<%= text_field_tag :last_name %>
+
+<%= label_tag :email %>
+<%= text_field_tag :email %>
+
+<%= label_tag :password %>
```
Comment (jpc20) [inline app/views/users/new.html.erb:14]: Is there a password tag we can use to hide the password when it is being entered?

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+require "rails_helper"
+
+
+
+describe User, type: :model do
+  describe "validation" do
+    it {should validate_presence_of(:first_name)}
+    it {should validate_presence_of(:last_name)}
+    it {should validate_presence_of(:email)}
+    it {should validate_presence_of(:password)}
```
Comment (jpc20) [inline spec/models/user_spec.rb:10]: Looks like these tests aren't passing on TravisCI, do we still need to migrate? Or is shoulda-matchers not working?

File: db/schema.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "users", force: :cascade do |t|
+    t.string "first_name"
+    t.string "last_name"
+    t.string "email"
+    t.string "password_digest"
+    t.string "password_confirmation"
```
Comment (jpc20) [inline db/schema.rb:23]: Does password confirmation need to be stored in the db? Or is this something we can remove?

File: spec/models/user_spec.rb (hunk 6 above)
Comment (jpc20) [reply spec/models/user_spec.rb:10]: Actually It looks like we need to add the shoulda-matchers config to the rails-helper.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (stellakunzang) [reply app/controllers/users_controller.rb:8]: In my opinion, a flash message going over the character limit for the method line isn't nearly as bad as an actual method that goes on and on and crosses the line. So I would say it is totally fine, but other people may have better suggestions!

File: app/views/users/new.html.erb (hunk 5 above)
Comment (stellakunzang) [reply app/views/users/new.html.erb:14]: `password_field_tag` hides the password (makes it look like `........`)

File: config/routes.rb (hunk 3 above)
Comment (stellakunzang) [reply config/routes.rb:7]: looks good to me also

File: db/migrate/20200721213935_create_user.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+class CreateUser < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
+      t.string :first_name
+      t.string :last_name
+      t.string :email
+      t.string :password_digest
+      t.string :password_confirmation
```
Comment (stellakunzang) [inline db/migrate/20200721213935_create_user.rb:8]: I didn't use a password_confirmation in any past projects. Is this something we need to save in the database?

File: db/schema.rb (hunk 7 above)
Comment (stellakunzang) [reply db/schema.rb:23]: oops, I made the same comment above

Comment (EricLarson2020) [reply db/schema.rb]: You correct, it is unnecessary, good catch!

File: app/views/users/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+<%= text_field_tag :password %>
+
+<%= label_tag :password_verification %>
+<%= text_field_tag :password_verification %>
```
Comment (stellakunzang) [inline app/views/users/new.html.erb:18]: I'm confused where we are confirming that the password and the password verification are matching. I've never used a verification before and I'm not sure how it all works. Can you clarify?

File: spec/models/user_spec.rb (hunk 6 above)
Comment (EricLarson2020) [reply spec/models/user_spec.rb:11]: Just fixed it, was missing a few lines in the rails_spec

Comment (stellakunzang) [reply spec/models/user_spec.rb:11]: It looks like the shoulda-matchers configure block isn't in rails_helper yet. I'll drop it below and @EricLarson2020 you can insert it and run the tests?

File: app/views/users/new.html.erb (hunk 5 above)
Comment (EricLarson2020) [reply app/views/users/new.html.erb:14]: added in a password field tag to hide password

**Pull request merged**